server/events/working_dir.go

```diff
@@ -232,6 +233,16 @@ func (w *FileWorkspace) forceClone(log logging.SimpleLogging,
 		baseCloneURL = w.TestingOverrideBaseCloneURL
 	}
 
+	var headerCmd []string
+
+	if headRepo.VCSHost.VcsToken != "" {
```

Comment:
Which vcs's return a vcstoken? Is it only azure devops?

Reply:
It is only AzureDevops - we still need to work out the best way to distinguish between cloud and on-premise versions. Some guidance on the preferred way here would be appreciated.

Reply:
Maybe we should pass in this value of a VcsToken then separately ?